src/System.Private.Interpreter/src/Internal/Runtime/Interpreter/ILInterpreter.cs

```diff
@@ -2645,5 +2656,71 @@ private void InterpretCall(int token)
             if (!signature.ReturnType.IsVoid)
                 _stack.Push(callInfo.ReturnValue);
         }
+        
+        private void InterpretNewObj(int token)
+        {
+            MethodDesc method = (MethodDesc)_methodIL.GetObject(token);
+            MethodSignature signature = method.Signature;
+            TypeDesc owningType = method.OwningType;
+
+            if (owningType.IsArray)
+            {
+                int[] lengths = new int[signature.Length];
+
+                for (int i = 0; i < signature.Length; i++)
+                {
+                    lengths[i] = PopWithValidation().AsInt32();
+                }
+
+                Array array = RuntimeAugments.NewMultiDimArray(owningType.GetRuntimeTypeHandle(), lengths, null);
```

Comment:
There are multiple variants of the multi-dimensional array constructors. See https://github.com/dotnet/runtime/blob/4f9ae42d861fcb4be2fcd5d3d55d5f227d30e723/src/coreclr/src/tools/Common/TypeSystem/Common/ArrayType.cs#L90

Reply:
Shoot. Forgot about jagged arrays 😄

Reply:
Hmm...just did a test and jagged arrays are interpreted okay. I guess I'm not sure what the ask here is. Please explain further?

Reply:
For example, `int[,]` array has two constructors. One that takes 4 arguments and one that takes 2 arguments. The one that takes 4 arguments is not handled correctly here.

You cannot call the one that takes 4 arguments directly in C#, but it is possible in IL.

Reply:
Looked at Ecma-335, that made things clearer. Updated the code to handle this scenario

Reply:
There are synthetic constructors for the jagged arrays as well. The reflection has good comments about it: https://github.com/dotnet/corert/blob/635cf21aca11265ded9d78d216424bd609c052f5/src/System.Private.Reflection.Core/src/System/Reflection/Runtime/TypeInfos/RuntimeArrayTypeInfo.cs#L91

Reply:
I just realized we have an existing helper for this that handles all the cases (gets used in the RyuJIT/CppCodegen/LLVM backend). Maybe we can just expose it through the `RuntimeAugments` class and call into that.

https://github.com/dotnet/corert/blob/635cf21aca11265ded9d78d216424bd609c052f5/src/System.Private.CoreLib/src/Internal/Runtime/CompilerHelpers/ArrayHelpers.cs#L17-L77

Reply:
Followed the suggestion of using `ArrayHelpers`